src/app.ts

```diff
@@ -0,0 +1,295 @@
+import assert from 'assert';
+import bodyParser from 'body-parser';
+import dotenv from 'dotenv';
+import express, { Request, Response, NextFunction, Application } from "express";
+import helmet from 'helmet';
+import plainJoi from 'joi';
+const joiZxcvbn = require('joi-zxcvbn');
+import sodium from 'libsodium-wrappers-sumo';
+import Mailgun from 'mailgun-js';
+import mongodb from 'mongodb';
+import Stripe, { IStripeError, customers, subscriptions, ICard } from 'stripe';
+import zxcvbn from 'zxcvbn';
+import * as http from 'http'
+
+dotenv.config();
+type Customer = customers.ICustomer;
+type Subscription = subscriptions.ISubscription;
+
+const token_window = 60000;
+const joi = plainJoi.extend(joiZxcvbn(plainJoi));
+
+// Mailgun Init
+const domain = 'mg.rhythmandala.com';
+const apiKey = process.env.MG_KEY;
+const mailgun = new Mailgun({ apiKey, domain });
+const emailBody1 = `<p>Dear Rhythm Aficionado,
+	<p>Thank you for subscribing to RhythMandala! Our goal is to make complex \
+rhythmic structures from all across the globe simple, accessible and playable \
+so that everyone can enjoy the benefits of understanding and experiencing \
+rhythm in the body. If you have any questions, please email ryan@tapgym.com.
+	<p>Go ahead and <a href='https://app.rhythmandala.com?`
+const emailBody2 = `'>click this link to complete your registration</a> and \
+have fun!`
+
+
+// Stripe Init
+const stripe = new Stripe(process.env.STRIPE_KEY);
+const plan = process.env.STRIPE_PLAN;
+
+// Mongo Init
+let users: mongodb.Collection;
+let server: mongodb.MongoClient;
+const mongodb_uri = process.env.MONGODB_URI;
+const mongoInit = async () => {
+	server = await mongodb.connect(mongodb_uri, { useNewUrlParser: true });
+	users = await server.db().collection('users');
+	await users.createIndex( { 'email': 1 }, { unique: true } );
+};
+
+// Express Init
+const app = express();
+const port = process.env.PORT;
+let expressServer: http.Server;
+app.use(helmet())
+app.use(bodyParser.json());
+
+// Start DB then Express then Mocha callback
+(async () => {
+	await mongoInit();
+	expressServer = await app.listen(port);
+	if (mocha_callback) {
```

Comment:
this `mocha_callback` stuff is a bit awkward. an alternative is to export an `async` `start` function which opens the mongo connection and then starts the server. then you could simply call that function at the beginning of your mocha tests. it would also be the function that gets called by your `index.js` (or whatever your entrypoint file is)